### PR TITLE
Adding new hash func support to Login

### PR DESCRIPTION
**Pull request by jskazinski**
- Added .env for storing test parameters
- Made sessionKey public so test function can access
- Update Login() to support two hash functions, without requiring caller changes
- Testing against older storage system and new storage system, but further local test suggested

### CODE REVIEW

File: main_test.go [hunk 1]
```diff
@@ -53,12 +63,12 @@ func TestReLoginFailed(t *testing.T) {
...
 
 	wrongClient.Password = "wrongpassword"
-	wrongClient.sessionKey = "outdated-session-key"
+	wrongClient.SessionKey = "outdated-session-key"
 
 	_, status, err := wrongClient.Request("/status/code/1")
 	g.Expect(err).NotTo(BeNil())
 	g.Expect(status.ResponseType).To(Equal("Error"))
-	g.Expect(status.Response).To(Equal("re-login failed"))
+	g.Expect(status.Response).To(Equal("request failed"))
```
Comment (paullaffitte) [inline main_test.go:71]: I don't understand the change here. It seemed fine before and the `re-login failed` error still exists in the code base.

File: main_test.go [hunk 2]
```diff
@@ -76,13 +86,13 @@ func TestInvalidXML(t *testing.T) {
...
 }
 
 func TestStatusCodeNotZero(t *testing.T) {
 	g := NewGomegaWithT(t)
 	_, status, err := client.Request("/status/code/1")
 
 	g.Expect(err).NotTo(BeNil())
-	g.Expect(status.ResponseTypeNumeric).To(Equal(1))
+	g.Expect(status.ResponseTypeNumeric).To(Equal(0))
```
Comment (paullaffitte) [inline main_test.go:97]: I don't understand the change here. Moreover, the function is called `TestStatusCodeNotZero`, which is the opposite of what is done.

File: request.go [hunk 3]
```diff
@@ -19,7 +19,8 @@ func (req *Request) execute(client *Client) ([]byte, int, error) {
 		return nil, 0, err
 	}
 
-	httpReq.Header.Set("sessionKey", client.sessionKey)
+	httpReq.SetBasicAuth(client.Username, client.Password)
```
[the remark this answers is not in the document]
Comment (jskazinski) [reply request.go:22]: We do not need the basic-auth here. I originally put it in as it is a way to work with both the older and newer apis, but then switched to using the new hash algorithm and did not remove this line. I will remove this change.

File: main_test.go (hunk 1 above)
Comment (jskazinski) [reply main_test.go:71]: This test did not pass for me on a newer system. Although this change works on newer system, it is clear that that the api responses changed and a better solution should be used. I will remove this change.

**Comment by jskazinski** [comment]
Hi Paul, I just read you original overall comment and will need to do some more work in order to follow your conventions. I will do some more review and get back to you.

**Comment by jskazinski** [comment]
Closing this request and starting over to address the review issues.

**Pull request closed without merging**